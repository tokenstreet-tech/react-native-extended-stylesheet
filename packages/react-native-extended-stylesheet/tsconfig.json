{
    "compilerOptions": {
        /* Visit https://aka.ms/tsconfig.json to read more about this file */

        /* Language and Environment */
        "target": "esnext",
        "lib": ["esnext"],
        "jsx": "react",

        /* Modules */
        "module": "esnext",
        "moduleResolution": "node",
        "baseUrl": "./",
        "paths": {
            "@tokenstreet/react-native-extended-stylesheet": ["./src/index"]
        },
        "resolveJsonModule": true,

        /* Emit */
        "importsNotUsedAsValues": "error",

        /* Interop Constraints */
        "esModuleInterop": true,

        "forceConsistentCasingInFileNames": true,

        /* Type Checking */
        "strict": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,

        "noImplicitReturns": true,
        "noFallthroughCasesInSwitch": true,
        "allowUnusedLabels": false,
        "allowUnreachableCode": false,
        "noImplicitUseStrict": false,
        "noStrictGenericChecks": false,

        /* Completeness */
        "skipLibCheck": true
    },
    "exclude": ["lib"]
}
