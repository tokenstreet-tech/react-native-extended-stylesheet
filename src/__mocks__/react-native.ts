import type ReactNative from 'react-native';
import type { DeepPartial } from 'utility-types';

const reactNativeMock: DeepPartial<typeof ReactNative> = {
    Dimensions: {
        get: () => ({ width: 100, height: 200, scale: 1, fontScale: 1 }),
    },
    Platform: {
        OS: 'ios',
        select: jest.fn(),
    },
    I18nManager: {
        isRTL: false,
    },
    StyleSheet: {
        create(obj: any) {
            return Object.keys(obj).reduce((res: any, key, index) => {
                res[key] = index;
                return res;
            }, {});
        },
        flatten(arr: any) {
            return arr.reduce((res: any, item: any) => Object.assign(res, item), {});
        },
        hairlineWidth: 1,
    },
};

module.exports = reactNativeMock;
