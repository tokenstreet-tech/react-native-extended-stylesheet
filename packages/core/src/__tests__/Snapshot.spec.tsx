import { EStyleSheet } from '@tokenstreet/react-native-extended-stylesheet';
import type { FC } from 'react';
import React, { useMemo } from 'react';
import { Platform, Text, View } from 'react-native';
import { create } from 'react-test-renderer';

describe('Snapshot', () => {
    it('should render correctly', () => {
        // Arrange
        EStyleSheet.build({
            $fontColor: '#F5FCFF',
        });

        const styles = EStyleSheet.create({
            // <-- use EStyleSheet instead of StyleSheet
            container: {
                flex: 1,
                justifyContent: 'center',
                alignItems: 'center',
                backgroundColor: '$fontColor', // <-- EStyleSheet variable
            },
            welcome: {
                fontSize: '1rem',
                textAlign: 'center',
                margin: 10,
            },
            instructions: {
                textAlign: 'center',
                color: '#333333',
                marginBottom: 5,
            },
        });

        const JestTestingScreen: FC = () => {
            const instructions = useMemo(
                () =>
                    Platform.select({
                        ios: `Press Cmd+R to reload,\nCmd+D or shake for dev menu`,
                        android: `Double tap R on your keyboard to reload,\nShake or press menu button for dev menu`,
                    }),
                []
            );

            return (
                <View style={styles.container}>
                    <Text style={styles.welcome}>Welcome to React Native!</Text>
                    <Text style={styles.instructions}>To get started, edit App.js</Text>
                    <Text style={styles.instructions}>{instructions}</Text>
                </View>
            );
        };

        // Act
        const tree = create(<JestTestingScreen />).toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });
});

/*
 * // Instead of creating __mocks__ you can build style for particular test in beforeAll hook:
 * import EStyleSheet from '@tokenstreet/react-native-extended-stylesheet';
 *
 * beforeAll(() => {
 *     EStyleSheet.build({
 *         $fontColor: '#F5FCFF',
 *     });
 * });
 */
