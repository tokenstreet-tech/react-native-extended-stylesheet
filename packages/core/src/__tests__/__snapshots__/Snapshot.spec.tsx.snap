// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should render correctly 1`] = `"View"`;
